compiler/rustc_mir_build/src/thir/pattern/usefulness.rs

```diff
@@ -1056,6 +1075,37 @@ impl<'tcx> Witness<'tcx> {
     }
 }
 
+/// Report that a match of a non_exhaustive type marked with `reachable_patterns` is not
+/// exhaustive enough.
+///
+/// NB: The partner lint for structs lives in `compiler/rustc_typeck/src/check/pat.rs`.
+fn non_exhaustive_reachable_match<'p, 'tcx>(
+    cx: &MatchCheckCtxt<'p, 'tcx>,
+    scrut_ty: Ty<'tcx>,
+    sp: Span,
+    hir_id: HirId,
+    witnesses: Vec<Pat<'tcx>>,
+) {
+    let joined_patterns = joined_uncovered_patterns(&witnesses);
+
+    cx.tcx.struct_span_lint_hir(NON_EXHAUSTIVE_REACHABLE_PATTERNS, hir_id, sp, |build| {
+        let mut lint = build.build("reachable patterns not covered of non exhaustive enum");
+        lint.span_label(sp, pattern_not_covered_label(&witnesses, &joined_patterns));
+
+        // IS THIS HELPFUL
+        // adt_defined_here(cx, &mut lint, scrut_ty, &witnesses);
```

Comment:
Some of the other missing pattern lints use this to add the enum definition site span to the warning, do we think this is worth adding?

Reply:
In our case it will always be defined outside the current crate so I don't think so